packages/integration-tests/src/test-projects/advanced-auth-and-functions/amplify/function.ts

```diff
@@ -24,6 +26,13 @@ export const funcNoMinify = defineFunction({
   },
 });
 
+export const funcProvided = defineFunction((scope) => {
+  return new NodejsFunction(scope, 'funcProvided', {
+    entry: './func-src/handler_provided.ts',
+    runtime: Runtime.NODEJS_18_X,
+  });
+});
```

Comment:
```suggestion
export const funcProvided = defineFunction((scope) => {
  return new NodejsFunction(scope, 'funcProvided', {
    entry: path.resolve(
      fileURLToPath(import.meta.url),
      '..',
      'func-src',
      'handler_provider.ts'
    ),
    runtime: Runtime.NODEJS_18_X,
  });
});
```


I've run e2e tests here https://github.com/aws-amplify/amplify-backend/actions/runs/12552245657 .

It seems that we need to resolve the entry path differently for provided function. The reason is that provided function entry resolution doesn't use (can't use) our implicit logic from [here](https://github.com/aws-amplify/amplify-backend/blob/22089b7a7321b8fc99e178a8dc777ccd77690e12/packages/backend-function/src/factory.ts#L273). Which is fine but we need to adjust test to accommodate it.

The reason this doesn't surface in "in memory tests" and only in e2e tests runs is that we don't include this project in https://github.com/aws-amplify/amplify-backend/tree/main/packages/integration-tests/src/test-in-memory (it's a gap in testing, so don't worry about this in this PR, I've manually ran proposed change to make sure it'll make e2e tests passing).